compiler/optimizer/VPHandlers.cpp

```diff
@@ -4883,13 +4881,19 @@ TR::Node *constrainArraylength(OMR::ValuePropagation *vp, TR::Node *node)
             uintptr_t array = knot->getPointer(kobj->getIndex());
             if (vp->comp()->fej9()->isClassArray(vp->comp()->fej9()->getObjectClass(array)))
                {
-               uintptr_t length = vp->comp()->fej9()->getArrayLengthInElements(array);
-               vp->replaceByConstant(node, TR::VPIntConst::create(vp, length), isGlobal);
-               return node;
+               isKnownObj = true;
+               lowerBoundLimit = vp->comp()->fej9()->getArrayLengthInElements(array);
+               upperBoundLimit = lowerBoundLimit;
```

Comment:
That's what I did initially.  Then thinking it over, I was worried that people might push to have the `J9_PROJECT_SPECIFIC` code pulled out of OMR rather than relocating it to another method in OMR's Value Propagation.  So that's why I left it where it is, with a plan to remove the `J9_PROJECT_SPECIFIC` code from here after https://github.com/eclipse-openj9/openj9/pull/14411 is merged.

Certainly moving the `J9_PROJECT_SPECIFIC` to `OMR::ValuePropagation::getArrayLengthLimits` would be simpler.

Darryl @0xdaryl, any thoughts?

Reply:
>with a plan to remove the J9_PROJECT_SPECIFIC code from here after https://github.com/eclipse-openj9/openj9/pull/14411 is merged.

That make sense. Sorry I forgot that you mentioned [on the above description](https://github.com/eclipse/omr/pull/6323#issue-1121021089) that line:4873 to line:4886 will be removed after https://github.com/eclipse-openj9/openj9/pull/14411 is merged. Thanks for the clarification! It looks good to me to keep them for this PR.

Reply:
Thanks!

Reply:
In general, I am always supportive (and appreciative) of efforts to remove J9-specific code from OMR.  It doesn't have to be all in one shot though, so if it makes it easier to land a PR with J9-specific code still in, but you have a plan to relocate shortly thereafter, then I am supportive of that approach.